libcloud/compute/drivers/dimensiondata.py

```diff
@@ -1826,6 +1826,22 @@ def ex_clone_node_to_image(self, node, image_name, image_description=None):
         response_code = findtext(result, 'result', GENERAL_NS)
         return response_code in ['IN_PROGRESS', 'SUCCESS']
 
+    def ex_clean_failed_deployment(self, node):
```

Comment:
take a look at the node_to_node_id stuff, that way someoen can also put in an STR that is a node id and this will still work as expected.  Most functions should have this.